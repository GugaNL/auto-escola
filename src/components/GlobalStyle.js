import { createGlobalStyle } from "styled-components"

const GlobalStyle = createGlobalStyle`
@import url("https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,300;0,700;1,600&display=swap");

html {
  font-family: "Poppins", sans-serif;
  font-weight: 300;
  font-size: 100%;
  color: #212121;
  box-sizing: border-box; //Garantir que todos os navegadores irão considerar o padding de acordo com o tamanho do objeto
}

*, *:before, *:after {
    box-sizing: inherit; //Faz com que herde o valor de cima, que no caso é o html
}

body {
    margin: 0;
}
`

export default GlobalStyle
