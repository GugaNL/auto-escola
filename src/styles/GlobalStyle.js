import React from "react"

//Styled component
import { createGlobalStyle } from "styled-components"

//Helmet
import { Helmet } from "react-helmet"

//Utils
import { BreakAt, BreakpointSize } from "./Breakpoints"

const GlobalStyle = createGlobalStyle`

html {
  font-family: "Poppins", sans-serif;
  font-weight: 300;
  font-size: 100%; //Padrão 16px
  color: #212121;
  box-sizing: border-box; //Garantir que todos os navegadores irão considerar o padding de acordo com o tamanho do objeto
}

/**, *:before, *:after {
    box-sizing: inherit; //Faz com que herde o valor de cima, que no caso é o html
}*/

body {
    margin: 0;
}

h1, h2, h3, h4, h5, h6 {
  line-height: 1.3;
  font-weight: 700;
  letter-spacing: 2px;
  margin-left: 0;
  
  strong {
    color: #ffc107;
  }
}

h1 {
  font-size: 2.5rem; //É o mesmo que 16px * 2.5 = 40px

  ${BreakAt(BreakpointSize.lg)} {
    font-size: 3.75rem;
  }
}

h2 {
  font-size: 2rem;

  ${BreakAt(BreakpointSize.lg)} {
    font-size: 3.125rem;
  }
}

h3 {
  font-size: 1.9rem;

  ${BreakAt(BreakpointSize.lg)} {
    font-size: 2.5rem;
  }
}

h4 {
  font-size: 1.3rem;
  font-weight: 600;

  ${BreakAt(BreakpointSize.lg)} {
    font-size: 2.125rem;
  }
}

h5 {
  font-size: 1.2rem;
  font-weight: 300;

  ${BreakAt(BreakpointSize.lg)} {
    font-size: 1.5rem;
  }
}

h6 {
  font-size: 1.1.rem;
  font-weight: 600;

  ${BreakAt(BreakpointSize.lg)} {
    font-size: 1.25rem;
  }
}
`

const GlobalStyleComposed = () => (
  <>
    <GlobalStyle />
    <Helmet>
      <meta charSet="utf-8" />
      <link rel="preconnect" href="https://fonts.gstatic.com" />
      <link
        href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;600;700&display=swap"
        rel="stylesheet"
      />
    </Helmet>
  </>
)

export default GlobalStyleComposed
