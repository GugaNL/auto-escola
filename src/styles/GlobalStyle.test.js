import React from "react"
import GlobalStyle from "./GlobalStyle"
import { render } from "@testing-library/react"

test("match snapshot", () => {
  render(<GlobalStyle />)

  expect(document.head).toMatchSnapshot()
})
