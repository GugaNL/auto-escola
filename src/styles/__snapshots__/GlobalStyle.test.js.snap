// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

@media (min-width:992px) {

}

<head>
  <style
    data-styled="active"
    data-styled-version="5.2.1"
  >
    html{font-family:"Poppins",sans-serif;font-weight:300;font-size:100%;color:#212121;box-sizing:border-box;}
    html body{margin:0;}
    html h1,html h2,html h3,html h4,html h5,html h6{line-height:1.3;font-weight:700;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;margin-left:0;}
    html h1 strong,html h2 strong,html h3 strong,html h4 strong,html h5 strong,html h6 strong{color:#ffc107;}
    html h1{font-size:2.5rem;}
    @media (min-width:992px){html h1{font-size:3.75rem;}}
    html h2{font-size:2rem;}
    @media (min-width:992px){html h2{font-size:3.125rem;}}
    html h3{font-size:1.9rem;}
    @media (min-width:992px){html h3{font-size:2.5rem;}}
    html h4{font-size:1.3rem;font-weight:600;}
    @media (min-width:992px){html h4{font-size:2.125rem;}}
    html h5{font-size:1.2rem;font-weight:300;}
    @media (min-width:992px){html h5{font-size:1.5rem;}}
    html h6{font-size:1.1.rem;font-weight:600;}
    @media (min-width:992px){html h6{font-size:1.25rem;}}
  </style>
</head>
`;
